doc/quickstart.rst

```diff
@@ -0,0 +1,187 @@
+   $ tar xzf charm-6.9.0.tar.gz
+
+To download the development version of Charm++, run:
+
+.. code-block:: console
+
+   $ git clone https://github.com/UIUC-PPL/charm
+
+
+To build Charm++, use the following commands:
+
+.. code-block:: console
+
+   $ cd charm
+   $ ./build charm++ netlrts-linux-x86_64 --with-production -j4
+
+This is the recommenced version to install Charm++ on Linux systems.
+For MacOS, substitute "linux" with "darwin". For advanced compilation options,
+please see Section :numref:`sec:install` of the manual.
+
+
+Parallel "Hello World" with Charm++
+-----------------------------------
+
+The basic unit of computation in Charm++ is a **chare**, which is a C++
+object. Chares have **entry methods** that can be invoked asynchronously.
+A Charm++ application consists of collections of chares (such as chare arrays)
+distributed among the processors of the system.
+
+Each Charm++ application consists of at least two files, a
+*Charm interface* (`.ci`) file, and a normal C++ file. The interface
+file describes the parallel interface of the application
+(such as chares, chare arrays, and entry methods), while the C++ files
+implement its behavior. Please see Section :numref:`programstructure`
+of the manual for more information about the program structure.
+
+In this section, we present a parallel *Hello World* example,
+consisting of the files ``hello.ci`` and ``hello.cpp``.
+
+
+The hello.ci file
+'''''''''''''''''
+
+The ``hello.ci`` file contains a mainchare, which starts and ends execution,
+and a ``Hello`` chare array, whose elements print the "Hello World" message.
+
+.. code-block:: charmci
+
+   mainmodule hello {
+     mainchare Main {
+       entry Main(CkArgMsg *m);
+       entry void done();
+     };
+     array [1D] Hello {
+       entry Hello();
+       entry void SayHi();
+     };            
+   };
+
+
+The hello.cpp file
+''''''''''''''''''
+
+The ``hello.cpp`` file contains the implementation of the mainchare and chare
+array declared in the ``hello.ci`` file above.
+
+.. code-block:: c++
+
+   #include "hello.decl.h" // created from hello.ci file above
+
+   /*readonly*/ CProxy_Main mainProxy;
+   constexpr int nElements = 8;
+
+   /*mainchare*/
+   class Main : public CBase_Main
+   {
+   public:
+     Main(CkArgMsg* m)
+     {
+       //Start computation
+       CkPrintf("Running Hello on %d processors with %d elements\n", CkNumPes(), nElements);
+       CProxy_Hello arr = CProxy_Hello::ckNew(nElements);
+       mainProxy = thisProxy;
+       arr[0].SayHi(0);
+     };
+
+     void done()
+     {
+       // Finish computation
+       CkPrintf("All done.\n");
+       CkExit();
+     };
+   };
+
+   /*array [1D]*/
+   class Hello : public CBase_Hello 
+   {
+   public:
+     Hello() {}
+
+     void SayHi()
+     {
+       CkPrintf("PE %d says: Hello world from element %d.\n", CkMyPe(), thisIndex);
+       if (thisIndex < nElements-1) {
+         thisProxy[thisIndex+1].SayHi(); // Pass the hello on
```

Comment:
A minor stylistic point, but I think the - and + operators should have spaces around them.

Reply:
Done